pkg/serverless/metrics/enhanced_metrics.go

```diff
@@ -240,6 +245,62 @@ func SendASMInvocationEnhancedMetric(tags []string, demux aggregator.Demultiplex
 	incrementEnhancedMetric(asmInvocationsMetric, tags, float64(time.Now().UnixNano())/float64(time.Second), demux, true)
 }
 
+type GenerateCPUEnhancedMetricsArgs struct {
+	UserCPUTimeMs   float64
+	SystemCPUTimeMs float64
+	Tags            []string
+	Demux           aggregator.Demultiplexer
+	Time            time.Time
+}
+
+// GenerateCPUEnhancedMetrics generates enhanced metrics for CPU time spent running the function in kernel mode,
+// in user mode, and in total
+func GenerateCPUEnhancedMetrics(args GenerateCPUEnhancedMetricsArgs) {
+	if strings.ToLower(os.Getenv(systemMetricsEnvVar)) == "false" {
+		return
+	}
```

Comment:
TODO: check with product if we should default enable/disable this feature

Reply:
on by default is great